src/Donations.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { Component } from 'react';
+import { formatCurrency, getDonations, Donation, track } from './utils';
+
+type State = {
+  loading: boolean;
+  donations: Donation[];
+  totalAmount: number;
+};
+
+class Donations extends Component {
+  state: State = {
+    loading: true,
+    donations: [],
+    totalAmount: 0,
+  };
+
+  async componentDidMount() {
+    const donations = this.state.donations;
+
+    const result = await getDonations();
+
+    for (var i = 0; i < result.length; i++) {
+      donations.push(result[i]);
+    }
+
+    this.setState({
+      donations: donations,
+      loading: false,
+      totalAmount: this.state.donations.reduce((p, c) => p + c.amount, 0),
+    });
+  }
+
+  renderDonation(donation: Donation) {
+    return (
+      <div>
+        <b style={{ textDecoration: 'underline' }}>
+          Donation amount:{' '}
+          {formatCurrency(donation.amount, donation.currencyCode)}
+        </b>
+        <span onClick={() => this.goToUrl(donation)}>Receipt</span>
+      </div>
+    );
+  }
+
+  goToUrl(donation: Donation) {
+    window.location.href = donation.receiptUrl;
+    track('receipt-click');
+  }
+
+  trackabout(e: any) {
+    e.preventDefault();
+    return track('about');
+  }
+
+  render() {
+    if (this.state.loading) {
+      return null;
+    }
+
+    return (
+      <div>
+        <div className="header">
+          <img src="/logo.png" />
+          <div className="nav">
+            <a href="/">
+              <img src="img/homeicon.svg" />
+            </a>
+            <a href="/" onClick={this.trackabout}>
+              <img src="img/aboutIcon.svg" />
+            </a>
+          </div>
+        </div>
+        <ul>{this.state.donations.map(this.renderDonation)}</ul>
```

Comment:
Need to add a `key` for a React list... somewhere! May require `renderDoantion` to be refactored into a component, so that it can be:
```suggestion
        <ul>{this.state.donations.map(donation => <Donation donation={donation} key={donation.receiptUrl} />)}</ul>
        Note - this is also made easier if there is an `id` for a donation available
```